azure/httpServer.ts

```diff
@@ -100,3 +111,121 @@ export class HttpServer extends pulumi.ComponentResource implements cloud.HttpSe
         });
     }
 }
+
+function createFactoryFunction(
+    createRequestListener: cloud.RequestListenerFactory): subscription.CallbackFactory<subscription.Context, any> {
+
+    return () => {
+        // First, setup the server.  This will only happen once when the module loads.
+        let server: http.Server;
+        try {
+            server = createServer(createRequestListener);
+        }
+        catch (err) {
+            // If we failed to create the server, set up a simple handler that just indicates
+            // the problem.
+            return context => {
+                context.log("Error occurred creating server: " + err.message + "\n" + err.stack);
+                context.done();
```

Comment:
note: these logs go into the Function's log stream (similar to cloudwatch).  It is not returned to the caller (i.e. a 3rd part) of this http endpoint..